RELEASING.md

```diff
@@ -26,6 +26,9 @@ To bump all (execpt the `/build` module's) dependencies run:
 
 4. Push the changes to upstream and create a Pull Request on GitHub.
 
+5. Create a Pull Request with documentation updates in
+   [splunk/public-o11y-docs](https://github.com/splunk/public-o11y-docs/tree/main/gdi/get-data-in/application/go)
```

Comment:
Can we bundle it with step no 1?

Reply:
https://github.com/signalfx/splunk-otel-go/pull/1951/commits/132027dff988881dc13750747eba3401bb98faec